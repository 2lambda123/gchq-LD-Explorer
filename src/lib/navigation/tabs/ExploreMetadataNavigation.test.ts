/* (c) Crown Copyright GCHQ */

import type { RoutedTabView } from './types';
import createTabDetail from './ExploreMetadataNavigation';

describe('ExploreMetadataNavigation', () => {
	let tabDetail: RoutedTabView<unknown>;

	beforeEach(() => {
		tabDetail = createTabDetail();
	});

	it('creates the correct config', () => {
		expect(tabDetail).toMatchSnapshot();
	});

	it('has the right type', () => {
		expect(typeof tabDetail).toBe('object');
	});
});
