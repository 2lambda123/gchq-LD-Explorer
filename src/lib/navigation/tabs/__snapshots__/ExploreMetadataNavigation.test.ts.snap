// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ExploreMetadataNavigation > creates the correct config 1`] = `
{
  "heading": "Metadata",
  "navigationLabel": "Metadata",
  "subheading": "Metadata is "data about the data". This can include things like labels, domains and ranges.",
  "tabs": [
    {
      "description": "English-language RDFS Labels within the active datasets.",
      "href": "/explore/metadata/labels",
      "tabIndex": 0,
      "title": "Labels",
    },
    {
      "description": "Domain assertions within the active datasets. Domains are used to infer the class of a subject based on its associated properties.",
      "href": "/explore/metadata/domains",
      "tabIndex": 1,
      "title": "Domains",
    },
    {
      "description": "Range assertions within the active datasets. Ranges are used to infer the class of an object based on its associated properties.",
      "href": "/explore/metadata/ranges",
      "tabIndex": 2,
      "title": "Ranges",
    },
  ],
}
`;
